docs/shared-content/structured-configuration-variables.include.md

```diff
@@ -0,0 +1,227 @@
+
+1. When you define the package [step](/docs/deployment-process/steps/index.md) in your [project](/docs/projects/index.md) enable the Structured Configuration Variables feature by clicking the **CONFIGURE FEATURES** link, selecting **Structured Configuration Variables** then clicking **OK**.
+2. In the **Features** section of the step template, specify the relative paths within the package to your Structured configuration files. For instance:
+
+```
+approot\packages\ASPNET.Core.Sample\1.0.0\root\appSettings.json
+```
+
+or
+
+```
+**/application.yaml
+```
+
+Octopus will find the target files and replace any matching configuration settings with the value of matching Octopus variables.
+
+### Selecting target files {#StructuredConfigurationVariablesFeature-SelectingTargetFiles}
+
+Target files are files that should have variable replacement applied to them. Multiple files can be supplied by seperating them with a new line. 
+
+There are a few options for supplying which files to use such as supplying a full path to the file, using a wild card to find all files in a directory, or using a wild card on the directory to find all files in that directory or deeper:
+
+**Specific file path**
+```
+ExampleProject/appSettings.json
+```
+
+**Match any .yaml files in root directory**
+```
+*.yaml
+```
+
+**Match any .json files in a specific directory**
+```
+Config\*.json
+```
+
+**Match any .yaml files in specified folder or deeper**
+```
+Application/**/*.yaml
+```
+
+The Target File field also supports [extended template syntax](/docs/projects/variables/variable-substitutions.md#VariableSubstitutionSyntax-ExtendedSyntax) which can allow conditionals and loops to be used to specify files. 
+
+### How the file type for target files is determined
+
+Structured Configuration Variables allows for replacement in both JSON and YAML files. To determine if a file is JSON or YAML, Calamari will first try and parse the file as JSON, and if it succeeds, it will treat the file as JSON. This is to allow backwards compatability from when this feature only supported replacement in JSON files. 
+
+If it doesn't parse as JSON, it will then use the file extension to determine if it should parse as YAML. If the file has the extension `yml` or `yaml`, Calamari will attempt to do variable replacement on the file as if it is a YAML file. 
+
+The type of the file can also be manually specified for cases when your file might not end in a file format this is automatically parsed. For example, if you have a YAML file called `application.config` that won't parse as YAML automatically as it doesn't end in `yml` or `yaml`. When specifying the file in the `Target Files` field, the following can be done to force the file to parse as YAML: `<yaml> application.config`. This also applies to json like the following: `<json> application.config`. 
```

Comment:
- [ ] If shipping without https://trello.com/c/u3zFa4M5, remove the description of file format override syntax

Reply:
Good call Roy, thanks! I've just removed it for now and we'll do any doco work around that with that ticket.